app/router/user.py

```diff
@@ -0,0 +1,168 @@
+async def create_user(request: Request):
+    """
+    This API writes user interaction details corresponding to a session ID.
+    """
+    data = await request.json()
+    query_params = {}
+    for key in data["form_data"].keys():
+        if key not in STUDENT_QUERY_PARAMS and key not in USER_QUERY_PARAMS:
+            raise HTTPException(
+                status_code=400, detail="Query Parameter {} is not allowed!".format(key)
+            )
+        query_params[key] = data["form_data"][key]
+    print(data)
+    if not data["id_generation"]:
+        if data["user_type"] == "student":
+            if (
+                "student_id" not in query_params
+                or query_params["student_id"] == ""
+                or query_params["student_id"] is None
+            ):
+                raise HTTPException(
+                    status_code=400, detail="Student ID is not part of the request data"
+                )
+            does_student_already_exist = student.verify_student(
+                student_id=query_params["student_id"]
+            )
+            if does_student_already_exist:
+                return query_params["student_id"]
+            else:
+                response = requests.post(user_db_url, params=query_params)
+                if response.status_code == 201:
+                    return query_params["student_id"]
+                raise HTTPException(status_code=500, detail="User not created!")
+    else:
+        if data["user_type"] == "student":
+            if (
+                "email" not in query_params
+                or query_params["email"] == ""
+                or query_params["email"] is None
+            ) or (
+                "phone" not in query_params
+                or query_params["phone"] == ""
+                or query_params["phone"] is None
+            ):
+                raise HTTPException(
+                    status_code=400,
+                    detail="Email/Phone is not part of the request data",
+                )
+            does_user_already_exist = get_users(
+                email=query_params["email"], phone=query_params["phone"]
+            )
+            if not does_user_already_exist:
+                while True:
+                    id = id_generation(data)
+                    does_student_already_exist = student.verify_student(student_id=id)
+                    if not does_student_already_exist:
+                        response = requests.post(user_db_url, params=query_params)
+                        if response.status_code == 201:
+                            return query_params["student_id"]
+                        raise HTTPException(status_code=500, detail="User not created!")
+
+            else:
+                response = student.get_students(does_user_already_exist["user_id"])
+                if response.status_code == 200:
+                    return response["student_id"]
+                raise HTTPException(status_code=500, detail="User not created!")
```

Comment:
just a doubt -- when exactly will this line occur?

so required user type is student, and id generation is set to true too
then we notice that the given id actually exists. 
so.. shouldn't student.get_students(student_id) return something tangible too?